gadgets/trace_open/test/trace_open_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019-2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"flag"
+	"io/fs"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/inspektor-gadget/inspektor-gadget/integration"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/command"
+	igrunner "github.com/inspektor-gadget/inspektor-gadget/pkg/testing/ig"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/match"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type Event struct {
+	eventtypes.CommonData
+
+	// Type indicates the kind of this event
+	Type string `json:"type"`
+
+	// Message when Type is ERR, WARN, DEBUG or INFO
+	Message string `json:"message,omitempty"`
+}
+
+type traceOpenEvent struct {
+	Event
+
+	MountNsID uint64      `json:"mountnsid"`
+	Pid       uint32      `json:"pid"`
+	Uid       uint32      `json:"uid"`
+	Gid       uint32      `json:"gid"`
+	Comm      string      `json:"comm"`
+	Ret       int         `json:"ret"`
+	Err       int         `json:"err"`
+	Flags     int         `json:"flags"`
+	FlagsRaw  int32       `json:"flagsRaw"`
+	Mode      int         `json:"mode"`
+	ModeRaw   fs.FileMode `json:"modeRaw"`
+	FName     string      `json:"fname"`
+	FullPath  string      `json:"fullPath"`
+}
```

Comment:
I think @mauriciovasquezbernal suggested this would be a better approach, I can't recall the reasoning for the same.
The mapping approach is already used by `run*` tests in `integration/inspektor-gadget`.

Reply:
> In the end, the goal of this type is to be translated to JSON to the be compared to the JSON output. I am wondering if we really need to define a type and if we should not rather use a map or JSON directly.

When I implemented the first test for the run gadget I had the following issue:
When decoding a JSON string into a `map[string]any` all numbers were of type `float32`. So in the test you would need to check the `Ret` variable with something like `jsonMap["Ret"] == 0.0f` or so.

Reply:
Please also consider [this](https://github.com/inspektor-gadget/inspektor-gadget/pull/2607#discussion_r1547713294).